### PR TITLE
Scrum 160 fe order details

### PR DESCRIPTION
**Pull request by asdelay**
## Describe your changes

- Created UI for OrderDetails Page
- Removed popups when user selects date where there is no route
- Modified header now you can go back with backArrow icon 
- Modified OrderItems now user can go to OrderDetail according to selected Order

## Issue ticket code (and/or) and link

- [SCRUM-160: FE order details](https://codecraftersintership.atlassian.net/browse/SCRUM-160)

### **General**
Screenshot 2024-12-19 at 00 37 21

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/OrderDetails/components/CollectionInformation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+const CollectioInformation: FC<CollectioInformationProps> = ({
+  collectionDate,
+  collectionTimeSlot,
+  collectionAddress,
+}) => {
+  return (
+    <div>
+      <StyledHeading variant='subtitle1'>
+        {' '}
```
Comment (oyakovytskyi) [inline src/pages/OrderDetails/components/CollectionInformation/index.tsx]: remove this?

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -15,9 +16,15 @@ interface HeaderProps {
 }
 
 const Header: FC<HeaderProps> = ({ pageName, hasBackIcon = false }) => {
+  const previousPage = -1;
+  const navigate = useNavigate();
```
Comment (oyakovytskyi) [inline src/components/Header/index.tsx:19]: local constant out of function or at least swap places with hook

File: src/pages/OrderDetails/components/CollectionInformation/index.tsx (hunk 1 above)
Comment (asdelay) [reply src/pages/OrderDetails/components/CollectionInformation/index.tsx]: fixed

File: src/components/Header/index.tsx (hunk 2 above)
Comment (asdelay) [reply src/components/Header/index.tsx:19]: fixed

**Review by oyakovytskyi — APPROVED** [review]

**Pull request merged**